diff --git a/programs/wineboot/wineboot.c b/programs/wineboot/wineboot.c
--- a/programs/wineboot/wineboot.c
+++ b/programs/wineboot/wineboot.c
@@ -1535,20 +1535,6 @@ static HANDLE start_rundll32( const WCHAR *inf_path, const WCHAR *install, WORD
     if (!(buffer = malloc( len * sizeof(WCHAR) ))) return 0;
     swprintf( buffer, len, L"%s setupapi,InstallHinfSection %s 128 %s", app, install, inf_path );
 
-    if (1)
-    {
-        /* CROSSOVER HACK bug 7736. Do prefix initialization in the root desktop. */
-        static WCHAR root[] = {'r','o','o','t',0};
-        HDESK desktop;
-
-        desktop = CreateDesktopW(root, NULL, NULL, 0, GENERIC_ALL, NULL);
-        if (desktop)
-        {
-            SetThreadDesktop(desktop);
-            CloseHandle(desktop);
-        }
-    }
-
     if (CreateProcessW( app, buffer, NULL, NULL, FALSE, 0, NULL, NULL, &si, &pi ))
         CloseHandle( pi.hThread );
     else
@@ -1557,46 +1543,6 @@ static HANDLE start_rundll32( const WCHAR *inf_path, const WCHAR *install, WORD
 
     free( buffer );
     return pi.hProcess;
-}
-
-/* ---------------------------------------------------------------
-**   This function is a CrossOver HACK for 9411 / 8979.
-** --------------------------------------------------------------- */
-static char* setup_dll_overrides(void)
-{
-    /* See CXBT.pm for reference */
-    static char overrides[] = "WINEDLLOVERRIDES=shdocvw=b;*iexplore.exe=b;advpack=b;atl=b;oleaut32=b;rpcrt4=b";
-    char* old_dlloverrides, *ret = NULL;
-    HANDLE hFile;
-
-    /* Save the original dll overrides so we can restore them after running
-     * rundll32.
-     */
-    old_dlloverrides = getenv("WINEDLLOVERRIDES");
-    if (old_dlloverrides)
-    {
-        ret = HeapAlloc( GetProcessHeap(), 0, sizeof("WINEDLLOVERRIDES=") + strlen(old_dlloverrides));
-        strcpy(ret, "WINEDLLOVERRIDES=");
-        strcat(ret, old_dlloverrides);
-    }
-
-    /* Check whether shdocvw is usable */
-    hFile = CreateFileA("c:/windows/system32/shdocvw.dll", FILE_READ_DATA,
-                        FILE_SHARE_READ | FILE_SHARE_WRITE,
-                        NULL, OPEN_EXISTING, FILE_ATTRIBUTE_NORMAL, NULL);
-    if (hFile != INVALID_HANDLE_VALUE)
-    {
-        char buf[0x40+20];
-        if (ReadFile(hFile, buf, sizeof(buf), NULL, NULL))
-        {
-            if (strncmp(buf+0x40, "Wine placeholder DLL", 20) != 0)
-                overrides[8] = 'd';
-        }
-        CloseHandle(hFile);
-    }
-    WINE_TRACE("for rundll32: %s\n", overrides);
-    putenv(overrides);
-    return ret;
 }
 
 static void install_root_pnp_devices(void)
@@ -1714,12 +1660,10 @@ static void update_wineprefix( BOOL force )
     {
         HANDLE process;
         DWORD count = 0;
-        char* old_dlloverrides = setup_dll_overrides();
 
         if ((process = start_rundll32( inf_path, L"PreInstall", IMAGE_FILE_MACHINE_TARGET_HOST )))
         {
-            /* HACK: Disable the wait window as it is deemed confusing */
-            HWND hwnd = 1 ? NULL : show_wait_window();
+            HWND hwnd = show_wait_window();
             for (;;)
             {
                 if (process)
@@ -1746,11 +1690,6 @@ static void update_wineprefix( BOOL force )
         update_user_profile();
 
         TRACE( "wine: configuration in %s has been updated.\n", debugstr_w(prettyprint_configdir()) );
-        if (old_dlloverrides)
-        {
-            putenv(old_dlloverrides);
-            free(old_dlloverrides);
-        }
     }
 
 done:
diff --git a/dlls/win32u/vulkan.c b/dlls/win32u/vulkan.c
--- a/dlls/win32u/vulkan.c
+++ b/dlls/win32u/vulkan.c
@@ -3003,6 +3003,7 @@ static void vulkan_init_once(void)
     const char *libvulkan, *graphics_backend;
 
     /* CW HACK 25909: Allow specifying libvulkan */
+#ifdef SONAME_LIBVULKAN
     if (!(libvulkan = getenv( "CX_LIBVULKAN" )) ||
         !(graphics_backend = getenv( "CX_ACTIVE_GRAPHICS_BACKEND" )) ||
         strcmp( graphics_backend, "wined3d" ))
@@ -3010,6 +3011,10 @@ static void vulkan_init_once(void)
     else
         ERR( "Using %s\n", libvulkan );
+#else
+    ERR( "Vulkan loader support is unavailable.\n" );
+    return;
+#endif
 
     if (!(vulkan_handle = dlopen( libvulkan, RTLD_NOW )))
     {
